crates/derive/src/types/batch.rs

```diff
@@ -0,0 +1,54 @@
+//! This module contains the enumerable [Batch].
+
+use super::batch_type::BatchType;
+use super::single_batch::SingleBatch;
+
+use alloy_rlp::{Decodable, Encodable};
+
+// TODO: replace this with a span batch
+/// Span Batch.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SpanBatch {}
+
+/// A Batch.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Batch {
+    /// A single batch
+    Single(SingleBatch),
+    /// Span Batches
+    Span(SpanBatch),
+}
+
+impl Decodable for Batch {
```

Comment:
Think the `Decodable` and `Encodable` traits are probably wrong for this type, given that `SpanBatch`es only contain components which are RLP encoded. Maybe add custom functions for this logic in a standard `impl Batch` block?

ref: https://github.com/ethereum-optimism/optimism/blob/develop/op-node/rollup/derive/channel_in_reader.go#L92-L118